src/magicbox/ResultPreviewsManager.ts

```diff
@@ -0,0 +1,212 @@
+import { $$, Dom } from '../utils/Dom';
+import { l } from '../strings/Strings';
+import { defaults, findIndex } from 'lodash';
+import { Component } from '../ui/Base/Component';
+import { Direction } from './SuggestionsManager';
+
+export interface IPopulateSearchResultPreviewsEventArgs {
+  suggestionText: string;
+  previewsQuery: Promise<ISearchResultPreview[]>;
+}
+
+export enum ResultPreviewsManagerEvents {
+  PopulateSearchResultPreviews = 'populateSearchResultPreviews'
+}
+
+export interface ISearchResultPreview {
+  element: HTMLElement;
+  onSelect: () => void;
+}
+
+export interface IResultPreviewsManagerOptions {
+  previewClass: string;
+  selectedClass: string;
+  previewHeaderText: string;
+}
+
+export class ResultPreviewsManager {
+  private options: IResultPreviewsManagerOptions;
+  private suggestionsPreviewContainer: Dom;
+  private resultPreviewsHeader: Dom;
+  private resultPreviewsContainer: Dom;
+  private lastPreviewsQuery: Promise<ISearchResultPreview[]>;
+  private lastSelectedSuggestion: HTMLElement;
+  private root: HTMLElement;
+
+  public get previewsOwner() {
+    return this.lastSelectedSuggestion;
+  }
+
+  public get hasPreviews() {
+    return !!this.suggestionsPreviewContainer;
+  }
+
+  public get focusedPreviewElement() {
+    if (!this.hasPreviews) {
+      return null;
+    }
+    const focusedElement = this.suggestionsPreviewContainer.findClass(this.options.selectedClass)[0];
```

Comment:
Could `this.suggestionsPreviewContainer.findClass(this.options.selectedClass)` be null/undefined ?

If yes, this would throw null ref.

Reply:
It technically shouldn't, since `this.hasPreviews` is only true if previews were appended to `suggestionsPreviewContainer`. However, I'll still add a check for the purpose of less confusing code.